### ISSUE TITLE
Customtask (custom task) is missing within Matomo

### ISSUE THREAD

**Issue by Chardonneaur** [piwik/piwik#20581]
Hi guys,

There is a great feature within Google Tag Manager named Customtask which allows to custom the data before sending it to Matomo.
Use case example, if we misspelled a campaign name and that we want to rewrite it before being collected into Matomo.
Here is a link explaining how powerful it is: https://www.simoahava.com/analytics/customtask-the-guide/

**Comment by bx80** [comment]
That sounds really useful @Chardonneaur :+1: 

I've assigned this for the product team to review and see where it fits into future plans.

**Comment by Chardonneaur** [comment]
@bx80 yes clearly seems like a feature to have in order to be an advanced and neat analytics software.

**Comment by sgiehl** [comment]
@Chardonneaur Should that one be part of the TagManager only or would that be more something you want to configure in the Matomo UI and Matomo itself would replace that whenever certain data comes in?

**Comment by Chardonneaur** [comment]
That's really a key question, Google has it within Google Tag Manager, but is it a good practice? I don't know

**Comment by heurteph-ei** [comment]
# MTM
## Advantages
Less work for the server
## Disadvantages
Not possible to custom data coming from other sources as server logs for example...

# In Matomo server side
## Advantages
Universal, whatever the tracking mean is used.
Can also centralize the data customization
## Disadvantages
Consumes server CPU
